### DISCUSSION TITLE
Coloured output is lost when running commands via ctx.run()

### DISCUSSION THREAD

**Discussion by ibc** [pyinvoke/invoke#976] [General]
When I run (for instance) `meson` commands using `ctx.run()`, the generated output lacks colors. I've found two related issues/PRs:

- https://github.com/pyinvoke/invoke/pull/663
- https://github.com/pyinvoke/invoke/pull/791

The former seems to be more related but it's not active since 2022. Is there any news about this?

**Comment by ibc** [comment]
By using `pty=True` in `ctx.run()` the problem is "solved" (obviously because `invoke` no longer obtains the output of the spawed process). However I'm not sure this is a good solution. If the executed command eventually requires user input (such as "you must accept Xcode licence", then I don't know if it will work.

**Reply by ibc** [reply]
Ok, I've tried this:

```python
ctx.run(
        f'read -p "Continue? (Y/N): " confirm && [[ $confirm == [yY] || $confirm == [yY][eE][sS] ]] || exit 1',
        echo=True,
        pty=True
);
```

and it works perfectly. So unless I'm wrong, using `pty=True` solves the lack of colored output problem.

**Reply by joaonc** [reply]
Note that `pty=True` doesn't work on Windows.   Mac and Linux only.
This error happens:
```
You indicated pty=True, but your platform doesn't support the 'pty' module!
```
